openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/Roles.spec.ts

```diff
@@ -0,0 +1,238 @@
+/*
+ *  Copyright 2024 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { expect, Page, test } from '@playwright/test';
+import { descriptionBox, redirectToHomePage, uuid } from '../../utils/common';
+
+const policies = {
+  dataConsumerPolicy: 'Data Consumer Policy',
+  dataStewardPolicy: 'Data Steward Policy',
+  organizationPolicy: 'Organization Policy',
+};
+
+const errorMessageValidation = {
+  ifPolicyNotSelected: 'Enter at least one policy',
+  ifNameNotEntered: 'Name size must be between 1 and 128',
+  lastPolicyCannotBeRemoved: 'At least one policy is required in a role',
+};
+
+const roleName = `Role-test-${uuid()}`;
+const description = `This is ${roleName} description`;
+
+const removePolicyFromRole = async (page: Page, policyName: string) => {
+  // Clicking on remove action for added policy
+  await page.locator(`[data-testid="remove-action-${policyName}"]`).click();
+
+  const modalText = await page.locator('.ant-modal-body').textContent();
+
+  expect(modalText).toBe(
+    `Are you sure you want to remove the ${policyName} from ${roleName}?`
+  );
+
+  await page.locator('[type="button"]:has-text("Confirm")').click();
+};
+
+const navigateToRolesPage = async (page: Page) =>
+  await page.goto('/settings/access/roles');
```

Comment:
@sweta1308 , can we use `settingClick` util here?

Reply:
Updated